trin-core/src/portalnet/storage.rs

```diff
@@ -427,7 +431,8 @@ impl PortalStorage {
 
         // Delete furthest data until our data usage is less than capacity.
         while self.capacity_reached()? {
-            // Unwrap because this was set in the block before this loop.
+            // Naked unwrap ok because this was set in the block before this loop.
+            #[allow(clippy::unwrap_used)]
             let id_to_remove = self.farthest_content_id.clone().unwrap();
```

Comment:
There are so many places this could go wrong in future code changes. Maybe it's fine to panic here, but I think an expect is clearer, and means we can skip the allow and comment. Something like:

```suggestion
            let id_to_remove = self.farthest_content_id.clone()
                .expect("farthest_content_id is always set immediately before this block");
```